### PR TITLE
Only adds  JS to home and artifacts

### PR DESCRIPTION
**Pull request by SaraVieira**
- Removes JS from any additional pages
- Introduces a `needs_js` bool

### CODE REVIEW

File: src/site/mod.rs [hunk 1]
```diff
@@ -48,9 +48,10 @@ impl Site {
         let dist = &config.dist_dir;
         Self::create_dist_dir(dist)?;
         for page in self.pages {
+            let js = page.needs_js.clone();
```
Comment (ashleygwilliams) [inline src/site/mod.rs]: is this clone necessary?

File: src/site/layout/mod.rs [hunk 2]
```diff
@@ -8,7 +8,7 @@ use crate::config::{analytics, theme, Config};
 use crate::errors::*;
 use axohtml::{html, text, unsafe_text};
 
-pub fn build(config: &Config, content: String) -> Result<String> {
+pub fn build(config: &Config, content: String, need_js: bool) -> Result<String> {
```
Comment (ashleygwilliams) [inline src/site/layout/mod.rs]: we should add needs_js to the Page struct, similar to "is index" and use it from there (also we should standardize on `needs_js`, theres 2 spellings in this PR)

**Review by ashleygwilliams — CHANGES_REQUESTED** [review]

File: src/site/layout/mod.rs (hunk 2 above)
Comment (SaraVieira) [reply src/site/layout/mod.rs]: my bad, changing everything to needs_js

File: src/site/mod.rs (hunk 1 above)
Comment (SaraVieira) [reply src/site/mod.rs]: apparently no, removed

**Review by ashleygwilliams — APPROVED** [review]
i definitely am gonna swoop in right after this and turn the 2 bools into an options struct for readability sake, but this gets the user facing feature right so approved ;)

**Comment by SaraVieira** [comment]
fixed the tests, waiting for green ci

**Pull request merged**